src/app/search/page.tsx

```diff
@@ -0,0 +1,20 @@
+import { getList } from '@/libs/microcms'
+import SearchPage from '@/components/pages/SearchPage'
+
+type Props = {
+  searchParams: {
+    q?: string
+  }
+}
+
+export default async function Page({ searchParams }: Props) {
```

Comment:
useSearchParams
https://nextjs.org/docs/app/api-reference/functions/use-search-params

>Using [useSearchParams()](https://nextjs.org/docs/app/api-reference/functions/use-search-params) in Client Components will skip static rendering and instead render all Client Components up to the nearest parent Suspense boundary on the client.

https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-rendering